tests/integration/resource_aware_pool_tests.cpp

```diff
@@ -0,0 +1,156 @@
+/////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-24, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include <stdio.h>
+
+#include <iostream>
+
+#include "camp/camp.hpp"
+#include "gtest/gtest.h"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Umpire.hpp"
+#include "umpire/config.hpp"
+#include "umpire/strategy/ResourceAwarePool.hpp"
+
+using namespace camp::resources;
+
+void host_sleep(int* ptr)
+{
+  int i = 0;
+  while (i < 1000000) {
+    int y = i;
+    y++;
+    i = y;
+  }
+  *ptr = i;
+  ptr++;
+}
+
+TEST(ResourceAwarePool_Host_Test, Check_States_Host)
+{
+  auto& rm = umpire::ResourceManager::getInstance();
+  auto pool = rm.makeAllocator<umpire::strategy::ResourceAwarePool>("rap-pool-host", rm.getAllocator("HOST"));
+
+  Resource r1{Host{}}, r2{Host{}};
+  int* ptr = static_cast<int*>(pool.allocate(r1, 1024));
+  int* compare_ptr1 = ptr;
+
+  EXPECT_EQ(getResource(pool, ptr), r1);
+  EXPECT_EQ(getPendingSize(pool), 0);
+
+  host_sleep(ptr);
+
+  pool.deallocate(r1, ptr);
+  // EXPECT_EQ(getPendingSize(pool), 0); // When only using host, there will be no pending chunks
```

Comment:
Can this check be enabled?

Reply:
No - currently whenever there is a deallocation, it always will first go into the pending state, regardless of what the underlying resource is. so technically that check is not true. however, the memory that is pending will have the event.check() return success and it will immidately be reused.. it's just a matter of my states not being updated fast enough to show it for a check.. I can explain that more if needed..

Reply:
After revising this from your comment below, I changed this and the check passes - so now if the event happens immediately (as in the case of the host resource) the do_deallocate is called and the chunk isn't even added to pending.

Reply:
I'm debating whether it makes sense to only do this check based on platform or not, certainly with the host resource you might as well do this.